scripts/docker/README.md

```diff
@@ -0,0 +1,169 @@
+
+> Advanced usage of the `docker-build-img.sh`
+```bash
+cd docker-images
+
+my_image/scripts/docker-build-img.sh --squash --cpus 4 --compress --
+
+# Above commandline example recognise the `--squash --cpus 4 --compress` parameters as
+# the `left` group (to the left of the standalone `--` parameter) and will be sent 
+# to the `docker build` command as IMMEDIATE_PARAMS (see bellow). The `right` group of
+# parameters will be empty (as there are no paremeters provided to the right of the
+# standalone `--` parameter) and will be set to the TAIL_PARAMS (= empty value), see
+# bellow.
+# Where general definition of the low level docker build commandline is:
+#   docker build $IMMEDIATE_PARAMS -t $DOCKER_IMAGE_TAG $TAIL_PARAMS $DOCKER_BUILD_CONTEXT_DIR
+# ==> And so the the resulting docker commandline (after substitution) will be:
+#   docker build --squash --cpus 4 --compress -t $DOCKER_IMAGE_TAG $DOCKER_BUILD_CONTEXT_DIR
+#, where DOCKER_IMAGE_TAG and DOCKER_BUILD_CONTEXT_DIR variables will be provided by 
+# scripts infrastructure (by `docker-env.sh` and `docker-env-common.sh`).
+```
+
+> Advanced usage of the `docker-run.sh`
+```bash
+cd docker-images
+
+my_image/scripts/docker-run.sh -p 8080:80 --cpus 4 -d -- build/examples/http_server -p 80
+
+# Above commandline example provides the `-p 8080:80 --cpus 4 -d` parameters to `docker run`
+# command as DOCKER_PARAMS, and the `build/examples/http_server -p 80` as the EXECUTABLE_PARAMS,
+# where low level docker run commandline is:
+#   docker run -i $USE_TTY -w $WORKDIR --rm $DOCKER_PARAMS -v $(pwd):$WORKDIR $DOCKER_IMAGE_TAG $EXECUTABLE_PARAMS
+# ==> And so the the resulting docker process commandline will be:
+#   docker run -i $USE_TTY -w $WORKDIR --rm -p 8080:80 --cpus 4 -d -t $DOCKER_IMAGE_TAG build/examples/http_server -p 80
+#, where values for USE_TTY, WORKDIR and DOCKER_IMAGE_TAG variables will be provided by
+# scripts infrastructure (by `docker-env.sh` and `docker-env-common.sh`)
+#
+# NOTE: The `-p 8080:80` parameter exposes internal docker container network port 80
+# to host OS network port 8080, what makes it externally accessible from host OS where
+# docker container is running. And `-d` param will ensure that docker runtime will
+# start the `build/examples/http_server` process in daemon mode (docker cli won't exit
+# until at least one daemon processes is running).
+```
+
+Docker <a name="docker_inst_setup"></a>
+=======================================
+
+The advantage of using the `Docker` is absolutely stable and reproducible setup (what is definitely important for test environment, but also for development and production environments) no matter what host OS setup is. Advantage is that this avoids necessity to install & maintain all necessary components directly on host OS (e.g. whole build chain, dependencies, tools, etc.) - this is done in docker and so it will eliminate issue with differencies of setup on different development machines, etc. ... 
```

Comment:
```suggestion
The advantage of using the `Docker` is absolutely stable and reproducible setup (what is definitely important for test environments, but also for development and production environments) no matter what host OS setup is. Advantage is that this avoids necessity to install & maintain all necessary components directly on host OS (e.g. whole build chain, dependencies, tools, etc.) - this is done in docker and so it will eliminate issue with differencies of setup on different development machines, etc. ... 
```